### PR TITLE
[DO NOT MERGE] - wip - golang + openssl support

### PR DESCRIPTION
**Pull request by lsm5**
- go mod edit -replace github.com/containers/image=github.com/containers/image@ddb11c0
- make vendor
- [ensure hack/tree_status.sh is happy] <- I notice this behaving
differently with go1.12.8 on ubuntu 18.04 v/s go1.12.9 on f30.

Signed-off-by: Lokesh Mandvekar <lsm5@fedoraproject.org>

@baude @mheon @mrunalp @rhatdan PTAL

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -10,12 +10,13 @@ require (
 	github.com/blang/semver v3.5.1+incompatible // indirect
 	github.com/buger/goterm v0.0.0-20181115115552-c206103e1f37
 	github.com/checkpoint-restore/go-criu v0.0.0-20190109184317-bdb7599cd87b
+	github.com/containerd/console v0.0.0-20181022165439-0650fd9eeb50 // indirect
 	github.com/containerd/continuity v0.0.0-20190426062206-aaeac12a7ffc // indirect
 	github.com/containernetworking/cni v0.7.1
 	github.com/containernetworking/plugins v0.8.1
-	github.com/containers/buildah v1.11.2
+	github.com/containers/buildah v1.9.0
```
Comment (mheon) [inline go.mod]: Eek? I think we downgraded pretty far here

**Comment by lsm5** [comment]
@vrothberg assuming I'm doing this right (PTAL..) c/image even prior to c/libtrust inclusion isn't good for vendoring in podman atm as per https://github.com/containers/libpod/pull/4128

File: go.mod (hunk 1 above)
Comment (lsm5) [reply go.mod]: recent updated commit avoids this, so maybe we're ok for now (tests still failing..)

**Comment by rhatdan** [comment]
Should this be done in Buildah first?

**Comment by vrothberg** [comment]
> @vrothberg assuming I'm doing this right (PTAL..) c/image even prior to c/libtrust inclusion isn't good for vendoring in podman atm as per #4128

The test failures look legit:
```
[+0988s] [It] podman push to oci-archive
[+0988s]   /var/tmp/go/src/github.com/containers/libpod/test/e2e/push_test.go:198
[+0988s] Running: /var/tmp/go/src/github.com/containers/libpod/bin/podman --root /tmp/podman_test002809880/crio --runroot /tmp/podman_test002809880/crio-run --runtime /usr/bin/runc --conmon /usr/libexec/podman/conmon --cni-config-dir /etc/cni/net.d --cgroup-manager systemd --tmpdir /tmp/podman_test002809880 --events-backend file --storage-driver vfs push docker.io/library/alpine:latest oci-archive:/tmp/podman_test002809880/alp.tar:latest
[+0988s] Getting image source signatures
[+0988s] Copying blob sha256:03901b4a2ea88eeaad62dbe59b072b28b6efa00491962b8741081c5df50c65e0
[+0988s] Error: Error copying image to the remote destination: Error creating an updated image manifest: Error preparing updated manifest: unsupported media type for compression: "application/vnd.docker.image.rootfs.diff.tar.gzip"
```

Will have a look at it ASAP.

**Comment by vrothberg** [comment]
Something went south here. Doing `go get github.com/containers/image@master` works for me but I can reproduce locally using this PR. I suggest redoing the vendor dance with `go get` instead of replacing the same repository with itself.

**Comment by lsm5** [comment]
> Something went south here. Doing `go get github.com/containers/image@master` works for me but I can reproduce locally using this PR. I suggest redoing the vendor dance with `go get` instead of replacing the same repository with itself.

Ack, there were a lot of downgrades when I did `go get...`. Specially buildah got downgraded from 1.11 to 1.9. I can try that again for confirmation.

**Comment by vrothberg** [comment]
> Ack, there were a lot of downgrades when I did `go get...`. Specially buildah got downgraded from 1.11 to 1.9. I can try that again for confirmation.

I faced that too and am unsure where it comes from. `go get` on the previous buildah version has worked locally.

**Comment by vrothberg** [comment]
Closing in favor of https://github.com/containers/libpod/pull/4165. Feel free to reopen if the change is still desired.

**Pull request closed without merging**